### PR TITLE
fix!: improve perf of model loading by reducing the scope of type ann…

### PR DESCRIPTION
**Pull request by tobymao**
…otations to just the select

### CODE REVIEW

File: sqlmesh/core/model/definition.py [hunk 1]
```diff
@@ -1009,7 +1009,10 @@ def columns_to_types(self) -> t.Optional[t.Dict[str, exp.DataType]]:
 
             self._columns_to_types = {
                 select.output_name: select.type or exp.DataType.build("unknown")
-                for select in annotate_types(query, schema=self._query_renderer.schema).selects
+                for select in (
+                    annotate_types(select, schema=self._query_renderer.schema)
+                    for select in query.selects
+                )
```
Comment (georgesittas) [inline sqlmesh/core/model/definition.py:1015]: I'm not sure if you can do this, don't you need the tables in scope to fetch the corresponding column types?

Comment (georgesittas) [reply sqlmesh/core/model/definition.py:1015]: Refer to this section https://github.com/tobymao/sqlglot/blob/main/sqlglot/optimizer/annotate_types.py#L339

Comment (tobymao) [reply sqlmesh/core/model/definition.py:1015]: we're only doing types for selects

Comment (georgesittas) [reply sqlmesh/core/model/definition.py:1015]: Yeah but I mean I'm not sure you can isolate the `annotate_types` call to just each select. For example if you don't have a proper query I think the scope creation will not work as expected and you won't annotate types?

**Pull request merged**